raiden/ui/web/src/app/components/channel-table/channel-table.component.ts

```diff
@@ -26,37 +25,43 @@ export class ChannelTableComponent implements OnInit {
     public tempChannel: Channel;
     public watchEvents: EventsParam[] = [];
     public tabIndex = 0;
+    private channelsSubject: BehaviorSubject<void> = new BehaviorSubject(null);
 
     constructor(
         private raidenConfig: RaidenConfig,
         private raidenService: RaidenService,
         private sharedService: SharedService,
-    ) { }
+    ) {
+    }
 
     ngOnInit() {
         let timeout;
         this.channels$ = this.channelsSubject.pipe(
             tap(() => clearTimeout(timeout)),
             switchMap(() => this.raidenService.getChannels()),
-            map((newChannels) =>
-                newChannels.map((newchannel) =>
-                    Object.assign(newchannel, { menu: null }) as WithMenu<Channel>
-                )
-            ),
+            map((newChannels) => newChannels.map((newchannel) =>
+            Object.assign(newchannel, {menu: null}) as WithMenu<Channel>
+        )),
             scan((oldChannels, newChannels) => {
                 // use scan and Object.assign to keep object references and
                 // improve *ngFor change detection on data table
                 for (const newchannel of newChannels) {
                     const oldchannel: WithMenu<Channel> = oldChannels.find((c) =>
                         c.channel_identifier === newchannel.channel_identifier);
-                    if (oldchannel && oldchannel.state == newchannel.state) {
-                        Object.assign(newchannel, { menu: oldchannel.menu });
+
+                    if (oldchannel && oldchannel.state === newchannel.state) {
+                        Object.assign(newchannel, {
+                            menu: this.updateMenu(newchannel, oldchannel.menu)
```

Comment:
These `oldchannel.menu` assignments were done to prevent a kind of flickering, where menu hide by itself, when the `channels` where updated (usually, each 5s), which made UX a little poorer, as user would need to click again to show the menu, but recently it was not working and kept flickering. May you, please, check and tell if your changes fixed it, or if not, also give a look on how to get it solved on this same PR, as you're going over this code?

Reply:
Initially tried to remove this `oldchannel.menu` and just updated the `MenuItem` entries each time, and this way I got the flickering, so I understood why this was done in such a way.

The PR didn't change anything in regards to flickering, but in my experience, there was no actual flickering before the PR, everything seemed to work.

Reply:
You are right though the menu indeed goes away when refreshing. I will look into it too.